tutorials/howtos/hide-index-page.md

```diff
@@ -0,0 +1,26 @@
+---
+title: "How can I disable an index page?"
+layout: tutorial
+---
+
+Occasionally you might want pieces that have their own show pages, but you don't need the index page. One case that's come up a few times is that pieces are listed in widgets across the site based on various criteria, but the organization doesn't have need for a central listing.
+
+Even if you don't create an `index.html` in your piece's pages module, one will exist by default anyway, likely looking completely unstyled. Bummer.
+
+For example, let's say we have a `cats` piece type. In my `cats-pages` module (extending `apostrophe-pieces-pages`), my `views` directory has a single `show.html` template file for the cat profile pages. So Oscar the cat can have his page at `example.com/cats/oscar`. If someone decides to try visiting `example.com/cats` we don't want that page to exist, but it will be there, listing cats, probably not looking that great.
+
+The solution is pretty simple! To get `example.com/cats` to return a 404 error (effectively no longer existing), we'll make a small addition to the [`beforeIndex`](../../modules/apostrophe-pieces-pages/index.html#before-index) method.
+
+In `lib/modules/cats-pages/index.js`, add the following to your `construct` method:
+
+```javascript
+construct: function (self, options) {
+  self.beforeIndex = function (req, callback) {
+    req.notFound = true;
+
+    return callback(null);
+  };
+}
+```
+
+See what we did there? Since `beforeIndex` runs before the index page is loaded, and by setting the request's `notFound` property to `true`, it'll return a 404 error rather than loading the page.
```

Comment:
In its current form this makes it hard to access the page to edit its settings, and if you make a new pieces-page you get redirected to a 404, both of which are weird experiences.

You should recommend:

`if (!req.data.page._edit) { req.notFound = true; }`

And adding something in `index.html` that says the index page is only visible to editors, in order to allow access to its page settings.

Reply:
I'll add that as an additional option. In some cases the site admins also shouldn't accidentally end up there, and of course it shouldn't be in the options for page types.